sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/StringUtilsSuite.scala

```diff
@@ -56,4 +56,32 @@ class StringUtilsSuite extends SparkFunSuite {
     assert(concat("1", "2") == "12")
     assert(concat("abc", "\n", "123") == "abc\n123")
   }
+
+  test("string concatenation with limit") {
+    def concat(seq: String*): String = {
+      seq.foldLeft(new StringConcat(7))((acc, s) => {acc.append(s); acc}).toString
+    }
+    assert(concat("under") == "under")
+    assert(concat("under", "over", "extra") == "underov")
+    assert(concat("underover") == "underov")
+    assert(concat("under", "ov") == "underov")
+  }
+
+  test("string concatenation return value") {
+    def checkLimit(s: String): Boolean = {
+      val sc = new StringConcat(7)
+      sc.append(s)
+      sc.atLimit
+    }
+    assert(checkLimit("under") == false)
+    assert(checkLimit("1234567"))
+    assert(checkLimit("1234567890"))
+  }
+
+  test("string concatenation append after limit") {
```

Comment:
How's this different from the test in L65?

Reply:
It's not anymore.  I will remove